tutorial/src/workspace/circuit.js

```diff
@@ -152,34 +214,40 @@ export default class Circuit extends Container {
 	 *   (i.e. they can be involved in 2q gates)
 	 */
 	_confirmFormat(qubits) {
-		this.isCompatible = true;
 		let numAncillas = 0;
-        qubits.forEach(qubit => {
-            if (qubit.role === 'a') {
```

Comment:
Can you make changes in this file: https://github.com/QCHackers/tqec/blob/main/frontend/src/plaquettes/CircuitWriter.js

Reply:
This PR is about frontend v2.

Reply:
@giangiac While individual experimentation is to be encouraged, we really should keep all of the core features in the same frontend, so that

* New and returning contributors have a single point of reference
* Work is not unnecessarily duplicated
* Bug fixes can benefit all users equally
* The public endpoint `tqec.app` is kept consistent

For example, take a look at https://github.com/QCHackers/tqec/blob/main/frontend/src/plaquettes/CircuitWriter.js. I wrote this file based on code you wrote, but made some changes to make it a little less brittle (using enumerated types instead of strings to track qubit types, for example)

Typically, release versions `v1, v2, v3...` are marked using [release tags](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository). Having a v1/v2... structure within the code itself is simply not good practice.

Also discussed some of these things with @afowler